### ISSUE TITLE
[Bug] Both Klein native weights (v1 & v2): CA layers are untrained (init-state export) AND key names mismatch v0.6.x code

### ISSUE THREAD

**Issue by tangjinghongart-create** [iFayens/ComfyUI-PuLID-Flux2#22]
# [Bug] Both Klein native weights (v1 & v2): CA layers are untrained (init-state export) AND key names mismatch v0.6.x code

## Summary

**Both** `pulid_flux2_klein_v1.safetensors` and `pulid_flux2_klein_v2.safetensors` (each 1.36 GB, HF commit `ef20fc7`) cannot inject identity through any code-side fix, because of **two independent defects**:

1. **The CA injection layers in both files are untrained** — they carry PyTorch default initialization values (verified by statistical fingerprint below, 12/12 CA layers in each file), so even a perfect key-mapping patch would inject init-state noise.
2. **Key naming/layer-count mismatch with v0.5.0+ code** — both files use pre-v0.5.0 names (`pulid_ca_double`/`pulid_ca_single`, 5+7 layers) while v0.5.0+ code expects `double_ca`/`single_ca` (10/30, now 12/60). `load_state_dict(strict=False)` silently drops all 84 CA keys, leaving all 504 CA parameters at initialization.

Both files were verified independently on this machine (byte-identical to the HF release: v1 sha256 `5fe643927aa398f0ffefa4f9796675c2c7d116e78aa96b2416d4d0f54c63fc10`, v2 sha256 `d5d291cb054eb6eceb25e3b46eff8f05f7b58f8f19a89ec76ba730a6ba8935bb`, each matching its repo LFS oid).

## Environment

- ComfyUI-PuLID-Flux2 @ `3a0a3f5` (current HEAD)
- Base model: FLUX.2 klein 9B (8 double + 24 single blocks, dim 4096)
- Result: face detection OK, `id_former` loads fine (35/35 keys), strength 0.8→1.4 produces no visible identity change at fixed seed. Same symptom as reported in #10 and #15.

## Evidence 1 — key accounting

Both v1 and v2 have the identical structure (119 keys):

```
Weight file keys: 119
  id_former.*: 35
  pulid_ca_double.*: 5 layers × 7 keys
  pulid_ca_single.*: 7 layers × 7 keys

Current code (PuLIDFlux2, dim=4096) expects: 539 keys
  id_former.*: 35 ✓ (all matched)
  double_ca.*: 12 layers (all missing → random init)
  single_ca.*: 60 layers (all missing → random init)

load_state_dict result: 35 matched / 504 missing / 84 unexpected (silently ignored)
```

Repro (any machine with the repo + weights):

```bash
python3 -c "
import sys, json, struct, torch
sys.path.insert(0, 'custom_nodes/ComfyUI-PuLID-Flux2')
import pulid_flux2
with torch.device('meta'):
    m = pulid_flux2.PuLIDFlux2(dim=4096)
mk = set(m.state_dict().keys())
with open('models/pulid/pulid_flux2_klein_v2.safetensors','rb') as f:
    n = struct.unpack('<Q', f.read(8))[0]
    wk = set(json.loads(f.read(n)))
print('matched:', len(mk&wk), 'missing:', len(mk-wk), 'unexpected:', len(wk-mk))
"
```

## Evidence 2 — the CA tensors are at init-state (not trained), in BOTH files

Statistical fingerprint of all 12 CA layers in v1 and in v2 (identical pattern in both):

| tensor | value | meaning |
|---|---|---|
| `pulid_ca_*.N.norm1.weight` | exactly **1.0**, std = 0.000000 | LayerNorm default init, zero gradient updates |
| `pulid_ca_*.N.norm1.bias` | exactly **0** | LayerNorm default init |
| `pulid_ca_*.N.to_q std` | **0.009018–0.009022** vs theory 0.009021 | matches kaiming_uniform theoretical bound/√3 for fan_in=4096 |
| `pulid_ca_*.N.to_q max` | exactly **0.015625** = ±1/√4096 | pinned to the init boundary, 12/12 layers, both files |

Controls:

- `id_former.*` (same files, loads and works): to_q max/init-bound = **2.09–2.62×** (v1) / **1.46–5.66×** (v2) → clearly trained
- `pulid_flux_v0.9.1.safetensors` (FLUX.1 PuLID, works on this machine): CA to_q max/init-bound = **3.6–10.0×**, norm1.weight std 0.013–0.041, norm1.bias max 0.037–0.074 → trained

Repro:

```bash
python3 -c "
import math
from safetensors import safe_open
for p in ['models/pulid/pulid_flux2_klein_v1.safetensors','models/pulid/pulid_flux2_klein_v2.safetensors']:
    with safe_open(p,framework='numpy') as f:
        for k in ['pulid_ca_double.0.to_q.weight','pulid_ca_single.6.to_q.weight',
                  'pulid_ca_double.0.norm1.weight','pulid_ca_double.0.norm1.bias',
                  'id_former.layers.0.to_q.weight']:
            a=f.get_tensor(k); b=1/math.sqrt(a.shape[-1])
            print(p.split('_')[-1], f'{k}: std={float(a.std()):.6f} theory={b/math.sqrt(3):.6f} max/bound={float(abs(a).max())/b:.4f}')
"
# Expected: CA to_q std == theory, max/bound == 1.0000; norm1.weight all-ones; norm1.bias all-zero
#           id_former to_q max/bound > 1.4 (trained)
```

## Evidence 3 — timeline (why this happened)

- v0.3.0/v0.4.0 (`75d3b5e`/`bac4308`): code used `pulid_ca_double/single`, 5/7 layers (v0.3.0) — both weight files' structure matches v0.3.0 exactly
- v0.5.0 (`2ed9899`, 2026-03-25): breaking rename to `double_ca`/`single_ca`, expanded to 10/30, **and removed the missing-keys warning** that v0.4.0's loader printed
- v0.6.0 (`43e17c3`): README announces "added native Klein weights v1 & v2", but the HF weights (commit `ef20fc7`) were never re-exported — they still carry the old names and (per Evidence 2) untrained CA values

Suggested fix on the author side: re-export the trained checkpoint with the v0.6.x key names (or ship a key-mapping in the loader), and restore the missing/unexpected warning that v0.4.0 had — this failure mode is currently completely silent.

## Impact

Every user following the current README (latest code + either HF Klein weight file) gets a PuLID that runs without errors but injects nothing. This matches #10 ("facial similarity is very poor") and #15 ("injection has no effect"). A silent strict=False load turns a hard failure into a soft, very hard-to-diagnose one.

Happy to provide any additional dumps/statistics if useful — and thank you for the work on this node.
